File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/backend/app/contracts/studentContract.py
from datetime import datetime
from typing import Literal
from uuid import UUID

from pydantic import BaseModel, ConfigDict, Field, field_validator


class StudentCreateContract(BaseModel):
    contractName: Literal["StudentCreate.v1"] = "StudentCreate.v1"
    fullName: str = Field(min_length=2, max_length=200)
    preferredName: str | None = Field(default=None, max_length=120)

    @field_validator("fullName", "preferredName")
    @classmethod
    def normalizeText(cls, value):
        if value is None:
            return None

        normalized = " ".join(value.split())
        return normalized or None


class StudentContract(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    contractName: Literal["Student.v1"] = "Student.v1"
    studentId: UUID
    fullName: str
    preferredName: str | None
    status: Literal["ACTIVE", "INACTIVE"]
    createdAt: datetime
    updatedAt: datetime

File: Documentacao/LIA2_008_PedagogicalAdaptive_Mobile_Thinking_FIX02/apps/studentWeb/src/contracts/studentContract.ts
export type StudentStatus = 'ACTIVE' | 'INACTIVE'

export interface StudentContract {
  contractName: 'Student.v1'
  studentId: string
  fullName: string
  preferredName: string | null
  status: StudentStatus
  createdAt: string
  updatedAt: string
}

export interface StudentCreateContract {
  contractName: 'StudentCreate.v1'
  fullName: string
  preferredName: string | null
}
